chips/nrf5x/src/ble_advertising_driver.rs

```diff
@@ -148,86 +226,223 @@ impl<'a, B, A> BLE<'a, B, A>
             app: container,
             kernel_tx: kernel::common::take_cell::TakeCell::new(tx_buf),
             alarm: alarm,
-            advertisement_interval: Cell::new(150 * <A::Frequency>::frequency() / 1000),
-            is_advertising: Cell::new(false),
-            // This keeps track of the position in the payload to enable multiple AD TYPES
-            offset: Cell::new(0),
         }
     }
 
+    fn initialize_advertisement_buffer(&self) {
+        for cntr in self.app.iter() {
+            cntr.enter(|app, _| {
+                app.advertisement_buf
+                    .as_mut()
+                    .map(|buf| {
+                        for i in buf.as_mut()[PACKET_START..PACKET_LENGTH].iter_mut() {
+                            *i = 0x00;
+                        }
+                        ReturnCode::SUCCESS
+                    })
+                    .unwrap_or_else(|| ReturnCode::EINVAL)
+            });
+        }
+    }
+
+
+    // TODO: Add "real" RNG
+    fn generate_random_address(&self, appid: kernel::AppId) -> ReturnCode {
+        self.app
+            .enter(appid, |app, _| {
+                app.advertisement_buf
+                    .as_mut()
+                    .map(|slice| {
+                        slice.as_mut()[PACKET_HDR_LEN] = 6;
+                        for i in slice.as_mut()[PACKET_ADDR_START..PACKET_ADDR_END + 1].iter_mut() {
+                            *i = 0xff;
```

Comment:
So is this currently just setting all addresses to `0xffffffffffff`? i.e., they'll all have the same address? A couple issues:

1. I believe this is an invalid address - the random part of both static and private random address may not be all 1s or all 0s (Vol. 6, Part B, 1.3.2).

2. The addresses of different apps should be differentiated, potentially by using the app id or some other signifier?

Reply:
Good point but I still want to use the TRNG for this but my plan is to go with `static device address`

*A static address is a 48-bit randomly generated address and shall meet the
following requirements*:
• The two most significant bits of the address shall be equal to 1
• At least one bit of the random part of the address shall be 0
• At least one bit of the random part of the address shall be 1

One suggestion is:
- byte 1       : 0x01 
- byte 2-5    (RNG)
- byte 6 :      0xFF

But for now I think using the Appid is good enough!

Reply:
yeah, that seems right to me